junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestExtension.java

```diff
@@ -91,6 +92,19 @@ public boolean supportsTestTemplate(ExtensionContext context) {
 		// @formatter:on
 	}
 
+	@SuppressWarnings("ConstantConditions")
+	private ArgumentsProvider getArgumentsProvider(Class<? extends ArgumentsProvider> clazz) {
+		try {
+			return ReflectionUtils.newInstance(clazz);
+		}
+		catch (Exception e) {
+			if (e instanceof NoSuchMethodException) {
+				throw new JUnitException(String.format("The constructor is not visible: %s", clazz.getName()), e);
```

Comment:
Please change the exception message to "Failed to find a no-argument constructor for ArgumentsProvider [%s]. Please ensure that a no-argument constructor exists and that the class is either a top-level class or a non-static nested class".

Reply:
Yes, you're of course right. Thanks for spotting that!